### PR TITLE
Run redis integration tests in Docker network

### LINKED ISSUE #262: Integration tests should use Docker networking instead of relying on host's networking
The redis integration tests connect to redis by going out of Docker, to the host machine, then back into Docker via the exposed port, using the IP address of the host machine.

This is fragile because it might break depending on the network setup of the host machine. A more reliable way would be to use Docker networks (and just connect to `redis:6379` inside the model container).

### PR DESCRIPTION
**Pull request by bfirsh**
The queue_controller server has been replaced by a separate container
that can run inside the network.

Closes #262
Closes #263

Signed-off-by: Ben Firshman <ben@firshman.co.uk>

### CODE REVIEW

File: test-integration/test_integration/util.py [hunk 1]
```diff
@@ -54,33 +57,23 @@ def docker_run(
...
-        "network",
-        "inspect",
-        "bridge",
-        "--format='{{ json (index .IPAM.Config 0).Gateway }}'",
-    ]
-    out, _ = subprocess.Popen(cmd, stdout=subprocess.PIPE).communicate()
-
-    return out.decode().strip().replace('"', "").replace("'", "")
+        pass
+        # subprocess.Popen(["docker", "rm", "--force", name]).wait()
```
Comment (zeke) [inline test-integration/test_integration/util.py:79]: did you mean to live this in here?

File: test-integration/test_integration/conftest.py [hunk 2]
```diff
@@ -1,25 +1,76 @@
...
+
+
+@pytest.fixture
+def upload_server_image():
+    """
+    Build the upload server once for the test run. The image doesn't change.
+    """
+    subprocess.run(
+        ["docker", "build", "-t", "cog-test-upload-server", "."],
+        cwd=Path(__file__).parent.parent / "upload_server",
```
Comment (zeke) [inline test-integration/test_integration/conftest.py:55]: TIL: `Path(__file__).parent.parent` -- I like how python uses dot notation for imports and paths.

**Review by zeke — APPROVED** [review]
Looks good!

File: test-integration/test_integration/util.py (hunk 1 above)
Comment (bfirsh) [reply test-integration/test_integration/util.py:79]: No -- good catch. Thank you!

**Comment by bfirsh** [comment]
@andreasjansson would appreciate some of your lovely eyeballs on this too, because you did previous implementation I think. A quick check just to make sure it makes conceptual sense with your knowledge of serving at least!

**Comment by bfirsh** [comment]
It's also worth noting that GitHub is hiding the bulk of this pull request behind an easy-to-miss "Load Diff" button because it's too large (sigh).

**Review by andreasjansson — APPROVED** [review]
Amazing, I love this. Better in every way, feels really good to get rid of that janky multiprocessing queue code.

**Comment by bfirsh** [comment]
Thanks for the 3rd and 4th eyes 🙌

**Pull request merged**